Sample NEM configuration and RPM packaging fixes

dpward opened the pull request:
Please pull these changes to fix DTD conformance of the sample NEM configurations, and to fix several issues with RPM package building and dependencies. These were tested on RHEL 6, Fedora 21, and Fedora 23 (still in development).

I was unable to reproduce any issue with autotools disabling compiler optimizations when the '-g' flag is present &mdash; I tried on Fedora 21 which has the same autoconf/automake version that you used, and I also tried in Fedora 23 (still in development).

sgalgano commented:
Thanks for the spec enhancements and XML cleanup.

I'm having an issue verifying the package documentation changes (https://github.com/dpward/emane/commit/3565c05ca73e921bc5a7d269e11f76f0c7347465). Testing was performed using a clean container (no previous emane packages) install  from rpm (Fedora 21) using:

```
# yum install *
```

The installation proceeds without issue but /usr/share/doc/emane is not populated:

```
# rpm -V emane
# ls /usr/share/doc/emane
ls: cannot access /usr/share/doc/emane: No such file or directory
```

Reinstalling a single RPM populates /usr/share/doc/emane:

```
# rpm -iv --replacepkgs emane-0.9.2-1.fc21.x86_64.rpm 
Preparing packages...
emane-0.9.2-1.fc21.x86_64
# ls /usr/share/doc/emane
AUTHORS  COPYING  ChangeLog  NEWS  PROBLEM-REPORT-FORM	README	SPONSOR
```

dpward commented:
Since you are installing in a Fedora 21 container, I think you might be seeing the effects of this change:
https://fedoraproject.org/wiki/Changes/Use_license_macro_in_RPMs_for_packages_in_Cloud_Image
What happens if you install the packages in a VM instead?

I updated the pull request to reflect this packaging change (in a way that is still backward-compatible with RHEL 6). I also added ownership of directories to RPMs. I tried these changes on Fedora 23, Fedora 21, and RHEL 6.

sgalgano commented:
Looks good. Thanks.
